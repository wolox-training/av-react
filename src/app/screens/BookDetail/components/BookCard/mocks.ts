export const OK_BOOK_RESPONSE_API = {
  id: 1,
  author: 'author',
  title: 'title',
  imageUrl: 'imageurl',
  editor: 'editor',
  year: 1900,
  genre: 'genre',
  currentRent: false
};
