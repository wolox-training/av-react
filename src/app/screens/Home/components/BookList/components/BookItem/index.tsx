import React from 'react';
import i18next from 'i18next';
import { useHistory } from 'react-router';

import { Book } from '~utils/types';
import { PATHS } from '~constants/paths';

import styles from './styles.module.scss';

interface BookProps {
  book: Book;
}

export default function BookItem({ book }: BookProps) {
  const history = useHistory();
  const bookClicked = (id: number) => {
    history.push(PATHS.bookDetail.replace(':id', `${id}`));
  };

  return (
    <div key={book.id} className={`${styles.book} column`} onClick={() => bookClicked(book.id)}>
      <img src={book.imageUrl} alt={i18next.t('BooksList:bookAlt') as string} className={styles.bookImage} />
      <div className={styles.bookTitle}>{book.title}</div>
      <div className={styles.bookAuthor}>{book.author}</div>
    </div>
  );
}
