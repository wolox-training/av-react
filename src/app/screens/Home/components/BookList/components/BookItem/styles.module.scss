@import 'variables/colors';

.book {
  box-shadow: 0 0 10px 6px $silver;
  height: 300px;
  width: 198px;

  .book-image {
    height: 200px;
    width: 142px;
    margin: 28px 28px 0;
  }

  .book-title {
    color: $black;
    font-size: 16px;
    font-weight: 700;
    line-height: 18px;
    margin: 12px 0 0 28px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    width: 109px;
  }

  .book-author {
    color: $gray;
    font-size: 14px;
    line-height: 18px;
    margin: 12px 0 0 28px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    width: 109px;
  }
}
