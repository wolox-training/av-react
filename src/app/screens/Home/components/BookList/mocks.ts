export const OK_BOOKS_API_RESPONSE = {
  count: 1,
  currentPage: 1,
  nextPage: 1,
  page: [
    {
      id: 1,
      author: 'test author',
      title: 'test title',
      imageUrl: 'test url',
      publisher: 'test publisher',
      year: 1960,
      genre: 'test genre',
      createdAt: '2020-10-10',
      updatedAt: '2020-10-10'
    },
    {
      id: 2,
      author: 'test author 2',
      title: 'test title 2',
      imageUrl: 'test url 2',
      publisher: 'test publisher 2',
      year: 1960,
      genre: 'test genre 2',
      createdAt: '2020-10-10',
      updatedAt: '2020-10-10'
    }
  ],
  totalCount: 1,
  totalPages: 1
};
