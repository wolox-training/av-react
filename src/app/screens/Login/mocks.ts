export const OK_LOGIN_API_RESPONSE = {
  data: {
    'access-token': 'test',
    data: {
      email: 'test@test.com',
      firstName: 'test',
      id: 1,
      lastName: 'test',
      locale: 'es'
    }
  }
};
