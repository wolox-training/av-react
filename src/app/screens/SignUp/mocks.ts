export const OK_SIGNUP_API_RESPONSE = {
  headers: {
    'access-token': 'test'
  },
  data: {
    email: 'test@test.com',
    firstName: 'test',
    id: 1,
    lastName: 'test',
    locale: 'es'
  },
  ok: true,
  originalError: null,
  problem: null,
  status: 201
};
