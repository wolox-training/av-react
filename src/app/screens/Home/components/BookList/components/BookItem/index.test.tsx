import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { OK_BOOKS_API_RESPONSE } from '../../mocks';

import BookItem from './';

const MAX_BOOKS_LENGTH = 1;

describe('Book test', () => {
  test('Should match snapshot', () => {
    const { container } = render(
      <Router>
        <BookItem book={OK_BOOKS_API_RESPONSE.page[0]} />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  test('Should render books data', async () => {
    render(
      <Router>
        <BookItem book={OK_BOOKS_API_RESPONSE.page[0]} />
      </Router>
    );
    await waitFor(() => {
      expect(screen.getAllByAltText('BooksList:bookAlt').length).toBe(MAX_BOOKS_LENGTH);
      expect(screen.getByText(OK_BOOKS_API_RESPONSE.page[0].title)).toBeInTheDocument();
      expect(screen.getByText(OK_BOOKS_API_RESPONSE.page[0].author)).toBeInTheDocument();
    });
  });
});
