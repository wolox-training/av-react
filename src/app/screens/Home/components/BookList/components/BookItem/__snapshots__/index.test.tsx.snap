// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem test Should match snapshot 1`] = `
<div>
  <div
    class="book column"
  >
    <img
      alt="BooksList:bookAlt"
      class="bookImage"
      src="test url"
    />
    <div
      class="bookTitle"
    >
      test title
    </div>
    <div
      class="bookAuthor"
    >
      test author
    </div>
  </div>
</div>
`;
